internal/worker/sshserver/server.go

```diff
@@ -91,6 +94,12 @@ func NewServerWorker(config ServerWorkerConfig) (worker.Worker, error) {
 	s.Server = &ssh.Server{
 		ConnCallback: s.connCallback(),
 		PublicKeyHandler: func(ctx ssh.Context, key ssh.PublicKey) bool {
+			// Delay the authentication to the embedded server.
+			// In Juju 3.6, public keys are associated with models rather than users,
+			// making it impossible to perform the authentication check at this point.
+			// Therefore, we delay the authentication until we instantiate the embedded server,
+			// at which point we have the destination model UUID.
+			ctx.SetValue(embeddedAuthenticationContextKey{}, true)
```

Comment:
I still prefer the naming `authenticateAgain`. Reading `embeddedAuthentication` isn't telling me directly what setting this to true is doing.